README.md

```diff
@@ -0,0 +1,32 @@
+# KCP API Catalog
+
+A library for kcp API catalog code.
+
+The API Catalog provides publicly available interfaces (APIs, code, CLIs) that allows users to curate a collection of available APIs (`APIExport`) in the KCP environment.
+
+## What is API Catalog?
+
+Each available API is encapsulated in a new API called `CatalogEntry` that is created by the user in a specific workspace.
+
+The `CatalogEntry` contains an identifier that links to its corresponding `APIExport`. Ideally, the identifier is a hash of `APIExport` name and its workspace location. This hash mechanism is utilized to conceal the direct identifier of `APIExport` which shouldn't be exposed to consumers for security reason. However, at the moment, `CatalogEntry` simply uses a list of `ExportReference` in the spec until the hash identifier is fully implemented.
+
+In addition of `APIExport` identifier information, the `CatalogEntry` have additional fields such as `Icons` and `Description` to allow users to add more information to represent the API specification and information for UI/Console usage.
+
+All of `CatalogEntry` in the same workspace is considered to be in the same group (catalog). The visibility of a `CatalogEntry` depends on the location of the workspace. For example, if the workspace is at `root` level, then the `CatalogEntry` is available for all tenants. If the workspace belongs to a specific organization such as `root:redhat`, then it will only be available to tenants under that `redhat` organization.
```

Comment:
What is the definition of "available" and "visible"?

Reply:
This is a bit of hazy topic given we never got a chance to discuss in the depth. My thought was that depending on the location of catalog workspace and the accessibility of that workspace, the users may see the list of catalog entries (via `list` command). That's the visibility concept. If they can see them, then they can bind them (via `bind` command). That is availability concept. Perhaps, we need a discussion to flesh it out fully.

Reply:
We could simplify this by assuming `list` would show catalogs present in current + all children's workspaces, and leaving the success of being able to bind or not upto the user. However in longer run it would be nice to differentiate between `available+can-bind` and `available+cannot-bind` catalogs based on user permissions (not just based on hierarchy of workspaces).